src/client/datascience/jupyter/debuggerVariableRegistration.ts

```diff
@@ -9,16 +9,20 @@ import { PYTHON_LANGUAGE } from '../../common/constants';
 import { IDisposableRegistry } from '../../common/types';
 import { Identifiers } from '../constants';
 import { IJupyterDebugService, IJupyterVariables } from '../types';
+import { IDebuggingManager } from '../../debugger/types';
+import { pythonKernelDebugAdapter } from '../../debugger/constants';
 
 @injectable()
 export class DebuggerVariableRegistration implements IExtensionSingleActivationService, DebugAdapterTrackerFactory {
     constructor(
         @inject(IJupyterDebugService) @named(Identifiers.MULTIPLEXING_DEBUGSERVICE) private debugService: IDebugService,
+        @inject(IDebuggingManager) private readonly debuggingManager: IDebuggingManager,
         @inject(IDisposableRegistry) private disposables: IDisposableRegistry,
         @inject(IJupyterVariables) @named(Identifiers.DEBUGGER_VARIABLES) private debugVariables: DebugAdapterTracker
     ) {}
     public activate(): Promise<void> {
         this.disposables.push(this.debugService.registerDebugAdapterTrackerFactory(PYTHON_LANGUAGE, this));
+        this.disposables.push(this.debuggingManager.registerDebugAdapterTrackerFactory(pythonKernelDebugAdapter, this));
```

Comment:
Seems like if you use the debugservice, registerDebugAdapterTrackerFactory, you wouldn't need the custom debug adapter tracker stuff.

Reply:
It would be a lot of work to do it that way. debugService is meant to be used for debugpy, as opposed to debuggingManager which uses debugpy in ipykernel.

It makes more sense this way. (And it also makes sense to remove debugService, stop using debugpy directly, and change the interactive window to also use ipykernel)

Reply:
Going to disagree. You're reinventing something that should already work.

DebugService works by multiplexing. It supports the old run by line and regular debugging. We actually don't need that anymore as it can just use the real VS code debug service instead.

Reply:
DebugService isn't used for debugpy. It's used for the old run by line. The debug cell path for the interactive window is just injecting debugpy and then basically doing the same thing you're doing in your kernel adapter. It does an attach to the kernel.

Reply:
I feel like the ownership is a bit muddled in my head as well from when I chatted with you David. The main core issue was just that debuggerVariables needed to be getting the debug messages incoming into it, right?

Reply:
So is the change here just detaching debuggerVariables from debug service and just hooking up to the vscode debug messages directly? Like we chatted about this would invalidate the old non-ipykernel 6.0 debugging functionality, but we might be ok with that at this point.

Reply:
First step was to use the debugAdapterTracker stuff instead of custom passing messages to the jupyter kernel and routing messages to the debug adapter's manually.

Second step (after this PR I think) would be to remove the JupyterDebugSession, MultiplexingDebugSession, and JupyterDebugger and just use what David has here for run by line/notebook debugging for interactive window debugging too.